packages/react/src/components/Table/Table.test.jsx

```diff
@@ -3050,4 +3050,167 @@ describe('Table', () => {
       expect(screen.queryByRole('button', { name: 'open and close list of options' })).toBeNull();
     });
   });
+
+  describe('batch actions', () => {
+    beforeEach(() => {
+      jest
+        .spyOn(HTMLElement.prototype, 'getBoundingClientRect')
+        .mockImplementation(() => ({ width: 100, height: 100 }));
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it('should fire callbacks for overflow batch actions', () => {
+      const rows = tableData.slice(0, 5);
+      const selectedIds = rows.map((row) => row.id);
+      const onApplyBatchAction = jest.fn();
+      render(
+        <Table
+          id="tableid1"
+          columns={tableColumns}
+          data={rows}
+          options={{ hasRowSelection: 'multi' }}
+          view={{
+            table: { selectedIds: selectedIds.slice(1, 5) },
+            toolbar: {
+              batchActions: [
+                {
+                  id: 'overflow-batch-action-text',
+                  labelText: 'overflow batch action text',
+                  isOverflow: true,
+                },
+                {
+                  id: 'test-overflow-batch-action-icon',
+                  labelText: 'overflow batch action with icon',
+                  renderIcon: Screen16,
+                  isOverflow: true,
+                },
+                {
+                  id: 'test-overflow-batch-action-hidden',
+                  labelText: "overflow batch action that's hidden",
+                  renderIcon: Screen16,
+                  isOverflow: true,
+                  hidden: true,
+                },
+                {
+                  id: 'test-overflow-batch-action-disabled',
+                  labelText: "overflow batch action that's disabled",
+                  renderIcon: Screen16,
+                  isOverflow: true,
+                  disabled: true,
+                },
+                {
+                  id: 'test-overflow-batch-action-delete',
+                  labelText: 'overflow batch action delete',
+                  renderIcon: TrashCan16,
+                  isOverflow: true,
+                  isDelete: true,
+                  hasDivider: true,
+                },
+              ],
+            },
+          }}
+          actions={{
+            toolbar: {
+              onApplyBatchAction,
+            },
+          }}
+        />
+      );
+
+      userEvent.click(screen.getByRole('button', { name: 'open and close list of options' }));
```

Comment:
nit: we could put the string 'open and close list of options' in a const called `overflowButtonName` or something since it is a bit of a weird and long name